cmd/routed-eni-cni-plugin/cni.go

```diff
@@ -311,17 +306,23 @@ func del(args *skel.CmdArgs, cniTypes typeswrapper.CNITYPES, grpcClient grpcwrap
 		return errors.Wrap(err, "del cmd: failed to load k8s config from args")
 	}
 
-	handled, err := tryDelWithPrevResult(driverClient, conf, k8sArgs, args.IfName, args.Netns, log)
-	if err != nil {
+	// Try to delete allocated network resources with data stored in PrevResult.
+	handled, skipIpamd, err := tryDelWithPrevResult(driverClient, conf, k8sArgs, args.IfName, args.Netns, log)
+	// On error, return immediately if pod was created with branch ENI. Otherwise, let IPAMD handle cleanup. Note that we cannot
+	// check conf.PodSGEnforcingMode as the pod we are deleting may have been added before the mode switch.
+	if err != nil && skipIpamd {
```

Comment:
what if `err != nil and !skipIpamd`, seems the error is ignored.

Reply:
The error still gets printed in `tryDelWithPrevResult`, but this is a case where we would still want to go through IPAMD for backwards compatibility. If we cannot delete from `PrevResult` and this pod was not created with a branch ENI (or we are not sure), then we give IPAMD a chance to do the delete.

In theory, this should never happen, but I figured we wanted to preserve backwards compatibility.

Reply:
The error is effectively ignored, as we are assuming that IPAMD would return the same error if there is a real error.